prover/src/bin/snarkify.rs

```diff
@@ -0,0 +1,48 @@
+use std::io;
+use std::rc::Rc;
+
+use async_trait::async_trait;
+use snarkify_sdk::prover::ProofHandler;
+use prover_runner::{
+    prover_core::Prover,
+    types::{Task, ProofDetail},
+    config::{Config, AssetsDirEnvConfig, SCROLL_PROVER_ASSETS_DIR_ENV_NAME},
+    version,
+};
+use scopeguard::defer;
+
+struct MyProofHandler;
+
+
+#[async_trait]
+impl ProofHandler for MyProofHandler {
+    type Input = Task;
+    type Output = ProofDetail;
+    type Error = String;
+
+    async fn prove(data: Self::Input) -> Result<Self::Output, Self::Error> {
+        let config: Config = Config::from_file("config.json".to_string()).map_err(|e| e.to_string())?;
+
+        if let Err(e) = AssetsDirEnvConfig::init() {
+            log::error!("AssetsDirEnvConfig init failed: {:#}", e);
+            std::process::exit(-2);
+        }
+
+        let prover = Prover::new(&config).map_err(|e| e.to_string())?;
+
+        log::info!(
+            "prover start successfully. name: {}, type: {:?}, publickey: {}, version: {}",
+            config.prover_name,
+            config.proof_type,
+            prover.get_public_key(),
+            version::get_version(),
+        );
```

Comment:
@cyphersnake, Need some advice here. We want the prover to be reused across different reqeusts, so we woud like it to be stored as a static variable and lazily initialized. Chatgpt recommended to use 
```
static PROVER: OnceCell<Prover> = OnceCell::new();

let prover = PROVER.get_or_init(|| {
  ...
})

```

but the compiler is giving many errors about thread-safty. In our use case we know that it is a single-threaded model, how should I rewrite the code to achieve my goal. Can you help take a look?

Reply:
https://doc.rust-lang.org/std/macro.thread_local.html

```rust
use std::cell::RefCell;

struct Prover;

impl Prover {
    fn new() -> Self {
        Self
    }

    fn ping(&self) -> u8 {
        42
    }

    fn pong(&mut self) -> u8 {
        42
    }
}

thread_local! {
    static GLOBAL: RefCell<Prover> = RefCell::new(Prover::new());
}

fn main() {
    GLOBAL.with_borrow(|v| {
        v.ping();
    });

    GLOBAL.with_borrow_mut(|v| {
        v.pong();
    });
}
```
Let me know if that's not enough

Reply:
Made the change as you suggested, and I think it works! Thanks!